src/wmtk/invariants/EnvelopeInvariant.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "Invariant.hpp"
+
+#include <memory>
+#include <wmtk/attribute/AttributeHandle.hpp>
+
+namespace fastEnvelope {
+class FastEnvelope;
+}
+
+namespace wmtk::invariants {
+class EnvelopeInvariant : public Invariant
+{
+public:
+    EnvelopeInvariant(
+        const Mesh& m,
+        const TypedAttributeHandle<double>& coordinate,
+        const Mesh& envelope_mesh,
```

Comment:
The mesh we want to put an envelope around could be non-manifold. As much as I dislike it, I think we should use IGL-style matrices here.